pysat/_instrument.py

```diff
@@ -1001,6 +1002,9 @@ def __setitem__(self, key, new_data):
                     # for list, array. Assume key[0] is integer
                     # (including list or slice).
                     self.data.loc[self.data.index[key[0]], key[1]] = new
+
+                self.meta._data_types[key[1]] = self.data[
+                    key[1]].values.dtype.type
```

Comment:
@aburrell, this breaks the icon ivm clean routines where a subset of data is replaced by np.nan.  I get 
```
TypeError: 'NoneType' object does not support item assignment
```

Reply:
Wrapping the meta update with `if not np.isnan(new):` fixes it.  Not sure if it makes sense to update if values are set to nan

Reply:
@jklenzing can you put together a bugfix PR?